### PR TITLE
add decoder for roll buy and roll sell

### PR DESCRIPTION
**Pull request by B-Naoufal**

### CODE REVIEW

**Review by Thykof — APPROVED** [review]
it can be nice to have a curl command to test that

File: pkg/node/sendoperation/buyrolls/buy_rolls.go [hunk 1]
```diff
@@ -47,3 +49,23 @@ func (b *BuyRolls) Message() []byte {
 
 	return append(msg, buf[:nbBytes]...)
 }
+
+func DecodeMessage(data []byte) (*OperationDetails, error) {
+	buf := bytes.NewReader(data)
+
+	// Read operationId
```
Comment (thomas-senechal) [inline pkg/node/sendoperation/buyrolls/buy_rolls.go:56]: For consistency with sell_roll
```suggestion
	// Skip operationId
```

File: pkg/node/sendoperation/buyrolls/buy_rolls.go [hunk 2]
```diff
@@ -47,3 +49,23 @@ func (b *BuyRolls) Message() []byte {
 
 	return append(msg, buf[:nbBytes]...)
 }
+
+func DecodeMessage(data []byte) (*OperationDetails, error) {
```
Comment (thomas-senechal) [inline pkg/node/sendoperation/buyrolls/buy_rolls.go:53]: This function looks exactly the same as the one in `sell_rolls`, shouldn't we move it in the `sendoperation` package to use the it at both places and avoid duplication ?

**Review by thomas-senechal — CHANGES_REQUESTED** [review]

File: pkg/node/sendoperation/buyrolls/buy_rolls.go (hunk 2 above)
Comment (Thykof) [reply pkg/node/sendoperation/buyrolls/buy_rolls.go:53]: if we move it, we should move it into a new file `rolls.go`

**Pull request merged**